backend/daemon/api/documents/v1alpha/comments.go

```diff
@@ -0,0 +1,343 @@
+package documents
+
+import (
+	"context"
+	"fmt"
+	"mintter/backend/backlinks"
+	documents "mintter/backend/genproto/documents/v1alpha"
+	"mintter/backend/pkg/errutil"
+	"mintter/backend/vcs"
+	"mintter/backend/vcs/hlc"
+	"mintter/backend/vcs/mttdoc"
+	"mintter/backend/vcs/sqlitevcs"
+	vcsdb "mintter/backend/vcs/sqlitevcs"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// CreateConversation implements the Comments server.
+func (api *Server) CreateConversation(ctx context.Context, in *documents.CreateConversationRequest) (*documents.Conversation, error) {
+	me, err := api.me.Await(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if in.DocumentId == "" {
+		return nil, errutil.MissingArgument("documentID")
+	}
+
+	docid, err := cid.Decode(in.DocumentId)
+	if err != nil {
+		return nil, errutil.ParseError("documentID", in.DocumentId, docid, err)
+	}
+
+	if err := validateSelectors("selectors", in.Selectors); err != nil {
+		return nil, err
+	}
+
+	if err := validateComment("initialComment", in.InitialComment); err != nil {
+		return nil, err
+	}
+
+	clock := hlc.NewClock()
+
+	perma, err := vcs.EncodePermanode(mttdoc.NewConversationPermanode(me.AccountID(), clock.Now()))
+	if err != nil {
+		return nil, err
+	}
+
+	conn, release, err := api.vcsdb.Conn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer release()
+
+	if err := conn.WithTx(true, func() error {
+		meLocal := conn.LookupIdentity(me)
+		convo := conn.NewObject(perma)
+		change := conn.NewChange(convo, meLocal, nil, clock)
+		batch := vcs.NewBatch(clock, me.DeviceKey().Abbrev())
+
+		batch.Add(vcs.RootNode, mttdoc.AttrConvDocument, docid)
+
+		for _, sel := range in.Selectors {
+			snode := vcs.NewNodeIDv1(time.Now())
+			rev, err := cid.Decode(sel.BlockRevision)
+			if err != nil {
+				return errutil.ParseError("selectors.blockRevision", sel.BlockRevision, rev, err)
+			}
+			batch.Add(snode, mttdoc.AttrConvSelectorBlockID, sel.BlockId)
+			batch.Add(snode, mttdoc.AttrConvSelectorBlockRevision, rev)
+			// If we have end on the selector we must have start, meaning that it's
+			// a fine-grained selection, not the whole block.
+			if sel.End != 0 {
+				batch.Add(snode, mttdoc.AttrConvSelectorStart, int(sel.Start))
```

Comment:
In case of fine grained selection do the above `batch.Add(snode, mttdoc.AttrConvSelectorBlockID, sel.BlockId)` and `batch.Add(snode, mttdoc.AttrConvSelectorBlockRevision, rev)` still make sense? if we have start and end, `blockID` and  `BlockRevision` are pointless since the `start` may be in one block and the `end` could be in another block, right?

Reply:
Start and end are per block. You can't have start in one block and end in the other block. But the conversation can have multiple Selectors pointing to multiple blocks. Having block revision allows to "apply" the selection to different versions of the document that might have this same block revision.

Does that answer your question, or I didn't get it?

Reply:
But fine grained would allow selecting a part of a block and a part of another right? In that case are you saying that there will be two selectors one of them `(StartSel_blk1, LastCharacter_blk1)` and the other `(FirstCharacter_blk2, EndSel_blk2)`

Reply:
@juligasa that's right: https://github.com/mintterteam/mintter/pull/1249/files#diff-1439186a501f1f28966dd0ed87d8688e206bddde4437c88ffa570799c8f12af1R38

So:

```
Selection = [Range]
Range = (BlockID, StartOffset, EndOffset)
```